apps/extension/client/src/checkTypegenNesting.ts

```diff
@@ -0,0 +1,166 @@
+import * as vscode from "vscode";
+
+const typegenPatternKeys = ["*.ts", "*.tsx", "*.mts", "*.cts"];
```

Comment:
q: are there any docs for what type of patterns are actually supported here by VSCode? I wonder if any other, more complex, patterns are supported 🤔

Reply:
I couldn't find any docs on this, I tried adding patterns like `*.ts*` but this was rejected by their schema

Reply:
I think the patterns are pretty simple and this should work OK.

We can take a look at the markdown description here:
https://github.com/microsoft/vscode/blob/1e60a4400f4fa00f65c6e212362153ec58716871/src/vs/workbench/contrib/files/browser/files.contribution.ts#L488

and valid substation types here:
https://github.com/microsoft/vscode/blob/d9225482faa9922efd0e88058c1bb9152d17a615/src/vs/workbench/contrib/files/common/explorerFileNestingTrie.ts#L224-L229

and patterns for keys are somewhat checked/processed here:
https://github.com/microsoft/vscode/blob/d9225482faa9922efd0e88058c1bb9152d17a615/src/vs/workbench/contrib/files/common/explorerFileNestingTrie.ts#L166-L185

This doesn't show the full picture but I think that basically only a single wildcard is allowed, other than that you can use an exact matches. So the current implementation looks good as no other "composite" patterns should be allowed.

Reply:
Wow, nice finds - thank you!